Gulpfile.js

```diff
@@ -396,13 +398,43 @@ gulp.task('build-website', ['prepare-website'], function (cb) {
         .on('exit', cb);
 });
 
-gulp.task('preview-website', ['build-website'], function (cb) {
-    connect()
-        .use('/testcafe', serveStatic('website/deploy'))
-        .listen(8080, function () {
-            opn('http://localhost:8080/testcafe');
-            cb();
-        });
+gulp.task('serve-website', ['build-website'], function (cb) {
+    var app = connect()
+        .use('/testcafe', serveStatic('website/deploy'));
+
+    server = app.listen(8080, cb);
+});
+
+gulp.task('preview-website', ['serve-website'], function () {
+    opn('http://localhost:8080/testcafe');
+});
+
+gulp.task('test-website', ['serve-website'], function (cb) {
+    var fail = false;
+    var brokenLinks = [];
+
+    var siteChecker = new blc.SiteChecker({ }, {
+        link: function (result) {
+            if (result.broken) {
+                brokenLinks.push(result.url.resolved ? result.url.resolved : result.url.original);
+                fail = true;
+            }
+        },
+        page: function (error, pageUrl) {
+            if (brokenLinks.length > 0) {
+                util.log('URL: ' + pageUrl);
+                util.log(util.colors.red('Broken links:'));
+                while (brokenLinks.length > 0)
+                    util.log(util.colors.red(brokenLinks.pop()));
+                util.log();
+            }
+        },
+        end: function () {
+            server.close(cb(fail ? 'Broken links found!' : void 0));
```

Comment:
the `server.close` function gets `callback` as an argument. But in fact you call it with the result of the `cb(...)` function execution (it's `undefined`). You should wrap `cb` calling with a function:
```js
server.close(function () {
    cb(fail ? 'Broken links found!' : void 0)
});
```